### DISCUSSION TITLE
Delay on sending messages

### DISCUSSION THREAD

**Discussion by jyothisjose** [AsamK/signal-cli#691] [Q&A]
We are using signal-rest-api for sending the messages with " --dbus " option.
Internally signal-rest-api calling the signal-cli command, we are getting a delay  of 3 - 8 sec while  sending the messages.
can you please let us know how can we reduce the send delay time to less than 1 sec.

**Comment by exquo** [comment]
Most of the time is spent in starting up Java virtual machine. 

You can bypass this by using a non-java command to send messages through dbus, e.g. `dbus-send`. See [wiki](https://github.com/AsamK/signal-cli/wiki/DBus-service#user-content-send-using-dbus-send)

There has recently been some work done to create a native binary that does not require JVM: #446
